src/_pytest/python_api.py

```diff
@@ -647,16 +687,15 @@ def raises(expected_exception, *args, match=None, **kwargs):
     for exc in filterfalse(
         inspect.isclass, always_iterable(expected_exception, BASE_TYPE)
     ):
-        msg = (
-            "exceptions must be old-style classes or"
-            " derived from BaseException, not %s"
-        )
+        msg = "exceptions must be derived from BaseException, not %s"
         raise TypeError(msg % type(exc))
 
     message = "DID NOT RAISE {}".format(expected_exception)
 
     if not args:
-        return RaisesContext(
+        # Type ignore because **kwargs is passed intentionally to trigger
+        # a TypeError if provided.
+        return RaisesContext(  # type: ignore
```

Comment:
I think we should instead:

```python
if kwargs:
    raise TypeError(...)
```

and not pass it on

Reply:
This was just changed in dfe54cd82f55f17f3c9e6e078325f306a046b93b, I left a question about it earlier here https://github.com/pytest-dev/pytest/pull/5529#discussion_r301934530

Reply:
yeah, let's do that here then :+1: